modules/llrt_crypto/src/subtle/mod.rs

```diff
@@ -34,13 +34,35 @@ pub enum Sha {
     Sha512,
 }
 
-fn get_sha(ctx: &Ctx<'_>, hash: &str) -> Result<Sha> {
-    match hash.to_ascii_uppercase().as_str() {
-        "SHA-1" => Ok(Sha::Sha1),
-        "SHA-256" => Ok(Sha::Sha256),
-        "SHA-384" => Ok(Sha::Sha384),
-        "SHA-512" => Ok(Sha::Sha512),
-        _ => Err(Exception::throw_message(ctx, "hash not found")),
+impl TryFrom<&str> for Sha {
+    type Error = &'static str;
+
+    fn try_from(hash: &str) -> std::result::Result<Self, Self::Error> {
+        match hash.to_ascii_uppercase().as_str() {
+            "SHA-1" => Ok(Sha::Sha1),
+            "SHA-256" => Ok(Sha::Sha256),
+            "SHA-384" => Ok(Sha::Sha384),
+            "SHA-512" => Ok(Sha::Sha512),
+            _ => Err("hash not found"),
```

Comment:
Which hash? Can be good to inform user what hash they try to use
Err(["\""_,"\" not available"].concat()